### PR TITLE
Access rights mgmt

### PR DESCRIPTION
**Pull request by tatut**
Added:
- user information to Datomic schema.
- new admin section
- role checks

### CODE REVIEW

File: app/backend/src/clj/teet/admin/admin_commands.clj [hunk 1]
```diff
@@ -0,0 +1,20 @@
+(ns teet.admin.admin-commands
+  (:require [teet.db-api.core :as db-api]
+            [datomic.client.api :as d]
+            [teet.user.user-roles :as user-roles]
+            [taoensso.timbre :as log]))
+
+(defn- new-user []
+  {:user/id (java.util.UUID/randomUUID)
```
Comment (outamaa) [inline app/backend/src/clj/teet/admin/admin_commands.clj]: That's some `teet.util/common/somesuch` material right there. Gotta have the `uuid` function. :)

File: app/backend/src/clj/teet/admin/admin_commands.clj [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+   :user/roles [:user]})
+
+(defmethod db-api/command! :admin/create-user [{conn :conn} user-data]
+  (let [user (merge (new-user)
+                    (select-keys user-data [:user/id :user/person-id :user/roles]))]
+    (log/info "CREATE USER: " user)
```
Comment (outamaa) [inline app/backend/src/clj/teet/admin/admin_commands.clj]: Here's a good `log/event` candidate in the future.

File: app/backend/src/clj/teet/admin/admin_queries.clj [hunk 3]
```diff
@@ -0,0 +1,12 @@
+(ns teet.admin.admin-queries
+  (:require [teet.db-api.core :as db-api]
+            [teet.user.user-roles :as user-roles]))
+
+(defmethod db-api/query :admin/list-users [{db :db} _]
+  {:query '[:find (pull ?e [*])
+            :where [?e :user/id _]]
+   :args [db]
+   :result-fn (partial mapv first)})
```
Comment (outamaa) [inline app/backend/src/clj/teet/admin/admin_queries.clj]: Is `(partial mapv first)` something that we want to do basically every time and then possibly something else as well?

File: app/backend/src/clj/teet/admin/admin_commands.clj [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    (d/transact conn
+                {:tx-data [user]})
+    :ok))
+
+(defmethod db-api/command-authorization :admin/create-user [{user :user} _]
```
Comment (outamaa) [inline app/backend/src/clj/teet/admin/admin_commands.clj]: I'd rather see the authorization closely tied to the command/query definitions than as multimethods. I guess we can have a convention of having the authorization defined next to the command/query definition, but I don't immediately see what additional value using multimethods adds here.

File: app/backend/src/clj/teet/environment.clj [hunk 5]
```diff
@@ -84,5 +84,5 @@
 
 (defn check-site-password [given-password]
   (let [actual-pw (config-value :auth :basic-auth-password)]
-    (and (some? actual-pw)
-         (= given-password actual-pw))))
+    (or (nil? actual-pw)
```
Comment (outamaa) [inline app/backend/src/clj/teet/environment.clj]: 👍

File: app/common/src/cljc/teet/user/user_roles.cljc [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  ([{roles :user/roles :as user} role-or-roles]
+   (cond
+     (nil? role-or-roles)
+     true
+
+     (keyword? role-or-roles)
+     (contains? roles role-or-roles)
+
+     :else
+     (some #(has-role? user %) role-or-roles))))
```
Comment (outamaa) [inline app/common/src/cljc/teet/user/user_roles.cljc]: Ok, so here the sematics are "needs to have at least one of the listed roles". Maybe update the docstring "Returns true if user has the *single* given role *or one of the multiple given roles*"

File: app/backend/src/clj/teet/admin/admin_commands.clj (hunk 4 above)
Comment (tatut) [reply app/backend/src/clj/teet/admin/admin_commands.clj]: Well that is an issue for another PR, the query/command authorization split is existing.

File: app/backend/src/clj/teet/admin/admin_queries.clj (hunk 3 above)
Comment (tatut) [reply app/backend/src/clj/teet/admin/admin_queries.clj]: Not all queries will `pull` maps, but it is a common one for sure.

**Pull request merged**